SignalServiceKit/protobuf/SignalService.proto

```diff
@@ -23,10 +23,14 @@ message Envelope {
     RECEIPT       = 5;
   }
 
+  // @required
```

Comment:
So far, I've stuck with using only a single "decorator', `@required`.

Reply:
Note that thanks to `@required`, I haven't need the "custom validation" logic at all so far.  It's still there in case we want to use it.

Reply:
Note that `@required` doesn't just address the "nullability" questions - we can also enforce it in the parsing process.

Reply:
`@required` have many effects, e.g.:

* The wrapper parsing methods will throw an exception if the value is not set. We use the same checking logic when parsing incoming protos and constructing outgoing protos, so this will affect both cases.
* Wrapper instances capture a non-optional reference to the value as a property, allowing other code to access with value without checking for nullability.
* Wrapper instances don't have a `has...` method.